app/src/main/java/tech/relaycorp/gateway/domain/courier/ProcessCargo.kt

```diff
@@ -1,30 +1,67 @@
 package tech.relaycorp.gateway.domain.courier
 
+import tech.relaycorp.gateway.common.Logging.logger
 import tech.relaycorp.gateway.data.disk.CargoStorage
+import tech.relaycorp.gateway.data.model.MessageAddress
+import tech.relaycorp.gateway.data.model.MessageId
 import tech.relaycorp.gateway.data.model.RecipientLocation
+import tech.relaycorp.gateway.domain.DeleteParcel
 import tech.relaycorp.gateway.domain.StoreParcel
+import tech.relaycorp.gateway.domain.StoreParcelAck
 import tech.relaycorp.relaynet.cogrpc.readBytesAndClose
 import tech.relaycorp.relaynet.messages.Cargo
+import tech.relaycorp.relaynet.messages.ParcelCollectionAck
+import tech.relaycorp.relaynet.messages.payloads.CargoMessage
+import java.util.logging.Level
 import javax.inject.Inject
 
 class ProcessCargo
 @Inject constructor(
     private val cargoStorage: CargoStorage,
-    private val readParcelsFromCargo: ReadParcelsFromCargo,
-    private val storeParcel: StoreParcel
+    private val readMessagesFromCargo: ReadMessagesFromCargo,
+    private val storeParcel: StoreParcel,
+    private val storeParcelAck: StoreParcelAck,
+    private val deleteParcel: DeleteParcel
 ) {
 
     suspend fun process() {
         val cargoes = cargoStorage.list()
         cargoes.forEach { cargoStream ->
             val cargo = Cargo.deserialize(cargoStream().readBytesAndClose())
```

Comment:
We should do `cargo.validate()` before any further processing.

Reply:
That was already done before storing the Cargo. We can do it again here, but it would be a bit wasteful.

Reply:
Sorry, I didn't realise. In that case disregard my comment.